src/components/file-browser/FileBrowser.ts

```diff
@@ -0,0 +1,54 @@
+import { useState } from 'react'
+import { div, h } from 'react-hyperscript-helpers'
+import FileBrowserDirectories from 'src/components/file-browser/FileBrowserDirectories'
+import FileBrowserFiles from 'src/components/file-browser/FileBrowserFiles'
+import FileBrowserProvider from 'src/libs/ajax/file-browser-providers/FileBrowserProvider'
+import colors from 'src/libs/colors'
+
+
+interface FileBrowserProps {
+  provider: FileBrowserProvider
+  title: string
+}
+
+const FileBrowser = ({ provider, title }: FileBrowserProps) => {
+  const [path, setPath] = useState('')
+
+  return div({ style: { display: 'flex', height: '100%' } }, [
+    div({
+      style: {
+        display: 'flex',
+        flexDirection: 'column',
+        width: 300,
+        height: '100%',
+        borderRight: `0.5px solid ${colors.dark(0.2)}`
+      }
+    }, [
+      div({
+        style: {
+          padding: '1rem 0.5rem',
+          borderBottom: `0.5px solid ${colors.dark(0.2)}`,
+          backgroundColor: colors.light(0.4)
+        }
+      }, [title]),
+      div({
+        style: {
+          flex: '1 0 0',
+          overflow: 'hidden auto',
+          background: '#fff'
+        }
+      }, [
+        h(FileBrowserDirectories, {
```

Comment:
I don't understand-- why is there both a `FileBrowserDirectories` and a `FileBrowserFiles` (down at line 49)? I am wondering if some documentation would help make it clearer how everything fits together.

Reply:
Ah, what was confusing me is that a directory includes files…. I thought FileBrowserFiles was like a child node inside of a directory.

Reply:
I could rename to something like FileBrowserDirectoryTree and FileBrowserFilesInDirectory?

Reply:
Maybe just `DirectoryTree` and `FilesInDirectory` or `DirectoryContents`? Everything is already in `file-browser`, so I wonder if if it's really useful to prefix every file with FileBrowser.

Reply:
👍  I tend to name files for navigation via fuzzy finder. In other parts of the code, we have four files all named List.js.

Reply:
I can see that, but here "FileBrowser" makes me think very specifically of the directory tree, which added to my confusion.

Reply:
Renamed in https://github.com/DataBiosphere/terra-ui/pull/3469/commits/7532cdb1ae394e50958754ce90ee32b56a542f76.